qcodes/instrument_drivers/tektronix/Keithley_3706A_Matrix_channels.py

```diff
@@ -0,0 +1,709 @@
+        firmware version and serial number.
+        """
+        switch_cards: List[Dict[str, str]] = []
+        for i in range(1, 7):
+            scard = self.ask(f'slot[{i}].idn')
+            if scard != 'Empty Slot':
+                model, mtype, firmware, serial = map(str.strip,
+                                                     scard.split(','))
+                sdict = {'slot_no': str(i), 'model': model, 'mtype': mtype,
+                         'firmware': firmware, 'serial': serial}
+                switch_cards.append(sdict)
+        return tuple(switch_cards)
+
+    def get_available_memory(self) -> Dict[str, Optional[str]]:
+        """
+        Returns the amount of memory that is currently available for
+        storing scripts, configurations and channel patterns.
+        """
+        memstring = self.ask('memory.available()')
+        system_memory, script_memory, \
+            pattern_memory, config_memory = map(str.strip, memstring.split(','))
+
+        memory_available: Dict[str, Optional[str]] = {
+            'System Memory  (%)': system_memory,
+            'Script Memory  (%)': script_memory,
+            'Pattern Memory (%)': pattern_memory,
+            'Config Memory  (%)': config_memory
+        }
+        return memory_available
+
+    def get_interlock_state(self) -> Tuple[Dict[str, str], ...]:
+        """
+        A function that collects the interlock status of the installed cards.
+        The channel relays can continue to operate even if the interlock
+        in the corresponding slot is disengaged, one cannot perform
+        measurements through the switching card, as the analog backplanes
+        cannot be energized.
+        """
+        slot_id = self._get_slot_id()
+        interlock_status = {0: 'Interlock is disengaged',
+                            1: 'Interlock is engaged'}
+        states: List[Dict[str, str]] = []
+        for i in slot_id:
+            state = self.ask(f'slot[{int(i)}].interlock.state')
+            state_dict = {'slot_no': i,
+                          'state': interlock_status[int(float(state))]}
+            states.append(state_dict)
+        return tuple(states)
+
+    def get_ip_address(self) -> str:
+        """
+        Returns the current IP address of the instrument.
+        """
+        return self.ask('lan.status.ipaddress')
+
+    def reset_local_network(self) -> None:
+        """
+        Resets the local network (LAN).
+        """
+        self.write('lan.reset()')
+
+    def save_setup(self, val: Optional[str] = None) -> None:
+        """
+        Saves the present setup.
+
+        Args:
+            val: An optional string representing the path and the file name
+                to which the setup shall be saved on a USB flash drive. If not
+                provided, the setup will be saved to the nonvolatile memory
+                of the instrument. Any previous saves shall be overwritten.
```

Comment:
"shall" sounds more like "wish/expectation" whereas the docstring should state the "fact", imho
```suggestion
                of the instrument, any previous saves will be overwritten.
```